cpp/src/arrow/flight/sql/odbc/CMakeLists.txt

```diff
@@ -36,7 +36,6 @@ endif()
 add_definitions(-DUNICODE=1)
 
 add_subdirectory(flight_sql)
```

Comment:
I suggest let's rename `flight_sql` folder to something like `odbc_impl` (or any suggestions from your side?). `flight_sql` is used in original Dremio code when ODBC was in its own repo. When arrow maintainers see `cpp/src/arrow/flight/sql/odbc/flight_sql` it might confuse them without the context.

Reply:
Renamed.